index.bs

```diff
@@ -10867,41 +10872,50 @@ There must exist an <dfn id="dfn-interface-prototype-object" export>interface pr
 for every non-callback [=interface=] defined,
 regardless of whether the interface was declared
 with the [{{NoInterfaceObject}}] [=extended attribute=].
-The interface prototype object for a particular interface
-has properties that correspond to the [=regular attributes=] and [=regular operations=]
-defined on that interface.
-These properties are described in more detail
-in sections [[#es-attributes]] and [[#es-operations]].
-
-As with the [=interface object=],
-the [=interface prototype object=] also has properties that
-correspond to the [=constants=] defined on that interface,
-described in [[#es-constants]].
-
-If the [{{NoInterfaceObject}}] extended attribute was not specified on the interface,
-then the [=interface prototype object=] must also
-have a <code class="idl">constructor</code> data property with attributes
-{ \[[Writable]]: <emu-val>true</emu-val>, \[[Enumerable]]: <emu-val>false</emu-val>, \[[Configurable]]: <emu-val>true</emu-val> }
-whose value is a reference to the [=interface object=] for the interface.
 
-<div algorithm="value of the [[Prototype]] internal slot">
+<div algorithm>
 
-    The [=interface prototype object=]
-    for a given interface |A| must have a
-    \[[Prototype]] [=internal slot=] whose value is returned from
-    the following steps:
-
-    1.  If |A| is declared with the [{{Global}}] [=extended attribute=], and
-        |A| [=support named properties|supports named properties=], then
-        return the [=named properties object=]
-        for |A|, as defined in [[#named-properties-object]].
-    1.  Otherwise, if |A| is declared to inherit from another
-        interface, then return the
-        [=interface prototype object=]
-        for the inherited interface.
-    1.  Otherwise, if |A| is declared with the [{{LegacyArrayClass}}]
-        extended attribute, then return {{%ArrayPrototype%}}.
-    1.  Otherwise, return {{%ObjectPrototype%}}.
+    The [=interface prototype object=] for a given [=interface=] |interface| and [=Realm=] |realm|
+    is <dfn lt="create an interface prototype object">created</dfn> as follows:
+
+    1.  Let |proto| be null.
+    1.  If |interface| is declared with the [{{Global}}] [=extended attribute=],
+        and |interface| [=support named properties|supports named properties=],
+        then set |proto| to the [=named properties object=] in |realm| of |interface|.
+    1.  Otherwise, if |interface| is declared to inherit from another interface,
+        then set |proto| to the [=interface prototype object=] in |realm|
+        of that [=inherited interface=].
+    1.  Otherwise, if |interface| is the {{DOMException}} [=interface=],
```

Comment:
Nice that we get to move this into here, instead of having it be an exception in the DOMException definition. We should remove the "must" from the custom bindings section, and instead make a statement of fact that refers back to this algorithm.

Reply:
Fixed in https://github.com/heycam/webidl/pull/494/commits/3c27f1d18ea6e8d248af369fc5ec9842ff084d35.